.github/workflows/csharp.yml

```diff
@@ -0,0 +1,116 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: C#
+
+on:
+  push:
+    paths:
+      - '.github/workflows/csharp.yml'
+      - 'ci/scripts/csharp_*'
+      - 'csharp/**'
+  pull_request:
+    paths:
+      - '.github/workflows/csharp.yml'
+      - 'ci/scripts/csharp_*'
+      - 'csharp/**'
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+
+  ubuntu:
```

Comment:
If the platforms are identical, could we consolidate them into a matrix argument?

Reply:
I'm totally cargo-culting this. Let's see if I can make that work.